### PR TITLE
fix command option completion

### PR DESCRIPTION
**Pull request by Milly**
- Remove deprecated command completion.
- Add completion for `:FernDo` and `:FernReveal`.
- Fix completion pattern matching.
- Add tests.

### CODE REVIEW

**Review by lambdalisue — CHANGES_REQUESTED** [review]
Thanks for your contribution 👍 
I just requested a small change so please check it.

File: autoload/fern/internal/complete.vim [hunk 1]
```diff
@@ -38,13 +40,13 @@ let s:options = {
       \}
 
 function! fern#internal#complete#opener(arglead, cmdline, cursorpos) abort
-  let pattern = '\V\^' . escape(a:arglead, '\\')
+  let pattern = escape(a:arglead, s:ESCAPE_PATTERN)
```
Comment (lambdalisue) [inline autoload/fern/internal/complete.vim:43]: The leading `^` is required.

```suggestion
  let pattern = '^' . escape(a:arglead, s:ESCAPE_PATTERN)
```

File: autoload/fern/internal/complete.vim [hunk 2]
```diff
@@ -38,13 +40,13 @@ let s:options = {
...
   let candidates = map(copy(s:openers), { _, v -> printf('-opener=%s', v) })
   return filter(candidates, { _, v -> v =~# pattern })
 endfunction
 
 function! fern#internal#complete#options(arglead, cmdline, cursorpos) abort
-  let pattern = '\V\^' . escape(a:arglead, '\\')
+  let pattern = escape(a:arglead, s:ESCAPE_PATTERN)
```
Comment (lambdalisue) [inline autoload/fern/internal/complete.vim:49]: Same

```suggestion
  let pattern = '^' . escape(a:arglead, s:ESCAPE_PATTERN)
```

**Review by lambdalisue — CHANGES_REQUESTED** [review]
One more

**Comment by Milly** [comment]
- Added tests for invalid patterns.

**Review by lambdalisue — APPROVED** [review]
LGTM

**Comment by lambdalisue** [comment]
Thanks a lot. Released as https://github.com/lambdalisue/fern.vim/releases/tag/v1.26.5

**Pull request merged**